sources/models/src/generator.rs

```diff
@@ -0,0 +1,258 @@
+//! The 'generator' module holds types that handles the settings generator metadata
+//! definition (containing command, strength and depth) among various systems
+//! like apiserver, sundog, datastore.
+//!
+//! The command field defines the command that needs to be executed to populate the
+//! setting.
+//! The strength field defines whether a setting needs to be deleted on reboot.
+//! The depth field defines how metadata is inherited across hierarchical levels,
+//! allowing a parent to provide metadata that can be applied at children at a given depth.
+//!
+//! SettingsGenerator type is used to hold generator that is applied strictly
+//! to the given setting and have depth 0.
+//! The RawSettingsGenerator holds the generators that can be dynamically applied
+//! to the successors at the given depth where a depth '0' means the generator
+//! should be applied on the given key.
+//!
+//! We use a custom deserializer because the metadata may not always be
+//! structured as an object; it can also appear as a string. This deserializer
+//! handles both formats, keeping the deserialization logic close to the struct
+//! for maintainability and clarity.
+
+use serde::{
+    de::{self, MapAccess, Visitor},
+    Deserialize, Deserializer, Serialize,
+};
+use serde_plain::derive_fromstr_from_deserialize;
+use std::fmt::{self, Display};
+
+/// Weak settings are ephemeral and deleted on reboot, regardless of whether or not it
```

Comment:
Weak settings aren't deleted on reboot, right? Just upgrade or downgrade?

```suggestion
/// Weak settings are ephemeral and deleted on upgrade or downgrade, regardless of whether or not it
```

Reply:
Migrator runs on reboot as well, so it deletes the weak settings on reboot as well.

Reply:
It runs, but that "nothing to do" message says that it isn't actually modifying the data store.

Reply:
I hadn't caught this before, but [we exit the process immediately when that message is emitted](https://github.com/bottlerocket-os/bottlerocket-core-kit/blob/523ed80b876714e7bdcb4629940d10058db43fd2/sources/api/migration/migrator/src/main.rs#L146).